langserve/server.py

```diff
@@ -971,159 +1019,162 @@ async def feedback(feedback_create_req: FeedbackCreateRequest) -> Feedback:
+        )
+        @app.post(
+            f"{namespace}/batch",
+            response_model=BatchResponse,
+            tags=route_tags,
+            name=_route_name("batch"),
+        )
+        async def _batch_docs(
+            batch_request: Annotated[BatchRequest, BatchRequest],
+            config_hash: str = "",
+        ) -> BatchResponse:
+            """Batch invoke the runnable with the given inputs and config."""
+            raise AssertionError("This endpoint should not be reachable.")
+
+        @app.post(
+            namespace + "/c/{config_hash}/stream",
+            include_in_schema=True,
+            tags=route_tags_with_config,
+            name=_route_name_with_config("stream"),
+        )
+        @app.post(
+            f"{namespace}/stream",
+            include_in_schema=True,
+            tags=route_tags,
+            name=_route_name("stream"),
+        )
+        async def _stream_docs(
+            stream_request: Annotated[StreamRequest, StreamRequest],
+            config_hash: str = "",
+        ) -> EventSourceResponse:
+            """Invoke the runnable stream the output.
 
-    @app.post(
-        namespace + "/c/{config_hash}/batch",
-        response_model=BatchResponse,
-        tags=route_tags_with_config,
-        name=_route_name_with_config("batch"),
-    )
-    @app.post(
-        f"{namespace}/batch",
-        response_model=BatchResponse,
-        tags=route_tags,
-        name=_route_name("batch"),
-    )
-    async def _batch_docs(
-        batch_request: Annotated[BatchRequest, BatchRequest],
-        config_hash: str = "",
-    ) -> BatchResponse:
-        """Batch invoke the runnable with the given inputs and config."""
-        raise AssertionError("This endpoint should not be reachable.")
+            This endpoint allows to stream the output of the runnable.
 
-    @app.post(
-        namespace + "/c/{config_hash}/stream",
-        include_in_schema=True,
-        tags=route_tags_with_config,
-        name=_route_name_with_config("stream"),
-    )
-    @app.post(
-        f"{namespace}/stream",
-        include_in_schema=True,
-        tags=route_tags,
-        name=_route_name("stream"),
-    )
-    async def _stream_docs(
-        stream_request: Annotated[StreamRequest, StreamRequest],
-        config_hash: str = "",
-    ) -> EventSourceResponse:
-        """Invoke the runnable stream the output.
+            The endpoint uses a server sent event stream to stream the output.
 
-        This endpoint allows to stream the output of the runnable.
+            https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events
 
-        The endpoint uses a server sent event stream to stream the output.
+            Important: Set the "text/event-stream" media type for request headers if
+                       not using an existing SDK.
 
-        https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events
+            This endpoint uses two different types of events:
```

Comment:
oh shoot - I added a `metadata` event here and didn't document it :/

Reply:
haha that's fine -- we can document in a bit